### PR TITLE
Add OpInfo for torch.equal and fix support for non-standard bools

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79393
* #79392
* #79391
* #79390
* __->__ #79389

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -11450,6 +11473,14 @@ def error_inputs_mean(op_info, device, **kwargs):
                     supports_fwgrad_bwgrad=True,
                     supports_two_python_scalars=True,
                     rhs_make_tensor_kwargs=dict(exclude_zero=True)),
+    OpInfo('equal',
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:11476]: Cool

**Review by mruberry — APPROVED** [review]
Let's just give the ciflow/trunk jobs a chance to run since this adds a new OpInfo (which is awesome!)

**Comment by peterbell10** [comment]
@mruberry PTAL. I had to add some more skips to get trunk ci passing, also added an `supports_tracing` variable to `OpInfo` since there were several tests where that was the reason it failed.

**Comment by ngimel** [comment]
@peterbell10 can you please manually rebase?

**Pull request closed without merging**